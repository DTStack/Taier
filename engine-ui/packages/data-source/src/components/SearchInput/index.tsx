/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useState } from 'react';
import { Input, Icon } from 'antd';

interface IProp {
	placeholder?: string;
	width?: number;
	onSearch: (value: string) => void;
	className?: string;
}
export default function SearchInput(props: IProp, any) {
	const { placeholder = '请求输入搜索内容', width = 200, onSearch, className } = props;

	const [value, setValue] = useState('');
	const IconSearch = () => (
		<span className="iconfont2 iconOutlinedxianxing_Search" data-testid="search-icon"></span>
	);
	return (
		<Input
			className={className}
			allowClear
			value={value}
			onChange={(e) => {
				setValue(e.target.value);
				if (!e.target.value) {
					onSearch(e.target.value);
				}
			}}
			onKeyDown={(e) => {
				if (e.key === 'Enter') {
					onSearch(value);
				}
			}}
			data-testid="input"
			placeholder={placeholder}
			suffix={
				<Icon
					component={IconSearch}
					onClick={() => {
						onSearch(value);
					}}
					style={{ cursor: 'pointer', color: '#999' }}
				/>
			}
			style={{ width: width }}
		/>
	);
}
