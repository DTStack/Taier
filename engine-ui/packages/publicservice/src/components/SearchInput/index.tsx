import React, { useState } from 'react';
import { Input, Icon } from 'antd';

export default function SearchInput({
  placeholder = '请求输入搜索内容',
  width = 200,
  onSearch,
}) {
  const [value, setValue] = useState('');
  const IconSearch = () => (
    <span className="iconfont iconOutlinedxianxing_Search"></span>
  );
  return (
    <Input
      allowClear
      value={value}
      onChange={(e) => {
        setValue(e.target.value);
        if (!e.target.value) {
          onSearch(e.target.value);
        }
      }}
      placeholder={placeholder}
      suffix={
        <Icon
          component={IconSearch}
          onClick={() => {
            onSearch(value);
          }}
          style={{ cursor: 'pointer', color: '#999' }}
        />
      }
      style={{ width: width }}
    />
  );
}
