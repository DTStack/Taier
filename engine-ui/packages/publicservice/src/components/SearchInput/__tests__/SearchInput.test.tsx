import React from 'react';
import SearchInput from '../index';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// const defaultProps = {
//   width: 200,
//   onSearch: (value) => {
//     console.log('value: ', value);
//   },
// };
let wrapper;
describe('test SearchInput', () => {
  //describe(name, fn)：描述块，讲一组功能相关的测试用例组合在一起
  //beforeEach(fn)在每个测试用例执行之前需要执行的方法
  //render，顾名思义，有助于渲染React组件
  // beforeEach(() => {
  //   wrapper = render(<SearchInput {...defaultProps}></SearchInput>);
  //   console.log('wrapper: ', wrapper);
  // });
  it('should take a snapshot', () => {
    wrapper = render(
      <SearchInput
        placeholder="请输入数据源名称或描述"
        width={200}
        onSearch={(value) => {
          console.log(value);
        }}
      />
    );

    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  // it('should get placeholder', () => {
  //   // expect(wrapper.getByTestClassName('ant-input')).
  //   expect(wrapper).toBeInstanceOf("ant-input")
  //  });

  afterEach(() => {
    cleanup(); //在每次测试后清理所有东西，以避免内存泄漏
  });
});
