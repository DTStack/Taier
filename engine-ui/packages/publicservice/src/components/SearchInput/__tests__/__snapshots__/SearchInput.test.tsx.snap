// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SearchInput should take a snapshot 1`] = `
<DocumentFragment>
  <span
    class="ant-input-affix-wrapper"
    style="width: 200px;"
  >
    <input
      class="ant-input"
      placeholder="请输入数据源名称或描述"
      type="text"
      value=""
    />
    <span
      class="ant-input-suffix"
    >
      <i
        aria-label="icon: search"
        class="anticon anticon-search"
        style="cursor: pointer; color: rgb(153, 153, 153);"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="search"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </i>
    </span>
  </span>
</DocumentFragment>
`;
